lib/hcm2/hcm2.js

```diff
@@ -28,8 +32,60 @@ const formatTime = datetime =>
 const formatDateTime = datetime =>
   formatDate(datetime) + ' ' + formatTime(datetime)
 
+/*
+ * NOTE: Even though there are TWO instances of the HCM2 in the Brain (as
+ * validator and as dispenser), this should be enough because (generally?)
+ * they're not used simultaneously.
+ */
+const rpcq = new Queue({ concurrency: 1, autostart: true, interval: 100 })
+
 const PORT = 8081
 const makeURL = endpoint => `http://localhost:${PORT}/api/${endpoint}`
+const rpc_call = (path, body) =>
+  rpcq.add(() => {
+    console.log("calling", path, "with body =", body)
+    return got.post(makeURL(path), {
+      json: true,
+      body: _.assign({ jsonrpc: '2.0', id: getID() }, body)
+    })
+    .then(resolve => ({ resolve }))
+    .catch(reject => ({ reject }))
+  })
+  .then(({ resolve, reject }) => reject ? Promise.reject(reject) : Promise.resolve(resolve))
+  .then(({ body: { error, result } }) => {
+    console.log(path, "error =", error)
+    console.log(path, "result =", result)
+    return error ? Promise.reject(error) : Promise.resolve(result)
+  })
```

Comment:
Long form for debugging reasons?

Reply:
What do you mean? `(...) => { ... }` ?

Reply:
I don't really get what's the point of this:


```js
    .then(resolve => ({ resolve }))
    .catch(reject => ({ reject }))
  })
  .then(({ resolve, reject }) => reject ? Promise.reject(reject) : Promise.resolve(resolve))
```

Reply:
ah that

The function given to the queue must always resolve, no matter what, otherwise it doesn't work properly. But we need/want to tell them apart on the outside. that's all that's doing -- passing the resolve/reject from inside the queue to the outside.